test/src/server/helpers/revisions.js

```diff
@@ -0,0 +1,418 @@
+
+		const formattedRevision = [
+			{
+				authorId: editorJSON.id,
+				entities: [],
+				revisionId
+			},
+			{
+				authorId: editorJSON.id,
+				entities: [],
+				revisionId: revisionId2
+			},
+			{
+				authorId: editorJSON.id,
+				entities: [],
+				revisionId: revisionId3
+			},
+			{
+				authorId: editorJSON.id,
+				entities: [],
+				revisionId: revisionId4
+			},
+			{
+				authorId: editorJSON.id,
+				entities: [],
+				revisionId: revisionId5
+			}
+		];
+		const revisionsWithEntities = await getAssociatedEntityRevisions(formattedRevision, orm);
+
+		expect(revisionsWithEntities.length).to.be.equal(5);
+		revisionsWithEntities.forEach((revision) => {
+			expect(revision.entities.length).to.be.equal(1);
+			revision.entities.forEach((entity) => {
+				expect(entity.defaultAlias.id).to.be.equal(expectedDefaultAliasId);
+			});
+		});
+	});
+});
+
+describe('getOrderedRevisionForEditorPage', () => {
+	// eslint-disable-next-line one-var
+	let editorJSON, req;
+	beforeEach(async () => {
+		const editor = await createEditor();
+		editorJSON = editor.toJSON();
+		const revisionAttribs = {
+			authorId: editorJSON.id,
+			createdAt: date.recent(),
+			id: 1
+		};
+		const promiseArray = [];
+		for (let id = 1; id <= 50; id++) {
+			revisionAttribs.id = id;
+			revisionAttribs.createdAt = date.recent();
+			promiseArray.push(
+				new Revision(revisionAttribs).save(null, {method: 'insert'})
+			);
+		}
+		await Promise.all(promiseArray);
+		req = {
+			app: {
+				locals: {
+					orm
+				}
+			},
+			params: {
+				id: editorJSON.id
+			}
+		};
+	});
+	afterEach(truncateEntities);
+
+	it('should throw an error for an invalid editor', async () => {
+		req.params.id = random.number();
+		try {
+			await getOrderedRevisionForEditorPage(0, 10, req);
+		}
+		catch (err) {
+			const expectedError = new error.NotFoundError('Editor not found', req);
+			expect(err.message).to.equal(expectedError.message);
+		}
+	});
+
+	it('should return sorted revisions with the expected keys', async () => {
+		const from = 0;
+		const size = 50;
+		const orderedRevisions = await getOrderedRevisionForEditorPage(from, size, req);
+		orderedRevisions.forEach(revision => {
+			expect(revision).to.have.keys(
+				'authorId',
+				'createdAt',
+				'editor',
+				'entities',
+				'notes',
+				'revisionId'
+			);
+		});
+		expect(orderedRevisions.length).to.equal(size);
+		expect(orderedRevisions).to.be.descendingBy('createdAt');
+	});
+
+	it('should return sorted revisions with notes sorted according to "postedAt"', async () => {
+		const revisionID = 51; // there are 50 revisions already created
+		const editor = await createEditor();
+		const editorJSON2 = editor.toJSON();
+		await new Revision({
+			authorId: editorJSON2.id,
+			createdAt: date.recent(),
+			id: revisionID
+		}).save(null, {method: 'insert'});
+
+		const noteAttrib = {
+			authorId: editorJSON2.id,
+			content: 'note content',
+			id: 1,
```

Comment:
Maybe safer to use a random.number() here and in the for loop below? What do you think?